### PR TITLE
MAINT: Split FIFF I/O into private submodule

### PR DESCRIPTION
**Pull request by larsoner**
First part is almost done I think -- just need to see what fails to tie up a few remaining loose ends that I'm sure CIs will reveal:

- [x] Move non-Raw parts of `mne/io` to `mne/_fiff`
- [x] Add deprecation wrappers for `mne/io/*.py` that moved
- [x] Add tests for deprecation wrappers
- [x] Get CIs green

The next phase will be a bit more work -- moving sibling MNE-* packages to use only public APIs if possible. This will probably require adding some new functions, but we'll see.

- [x] Check mne-bids: https://github.com/mne-tools/mne-bids/pull/1165
- [x] Check mne-bids-pipeline: None
- [x] Check mne-connectivity: https://github.com/mne-tools/mne-connectivity/pull/145
- [x] Check mne-qt-browser: https://github.com/mne-tools/mne-qt-browser/pull/191
- [x] Check mne-nirs: https://github.com/mne-tools/mne-nirs/pull/521
- [x] Check mne-realtime: https://github.com/mne-tools/mne-realtime/pull/54
- [x] Check mne-kit-gui: https://github.com/mne-tools/mne-kit-gui/pull/21
- [x] Check autoreject: https://github.com/autoreject/autoreject/pull/310

Then:

- [ ] Merge
- [ ] in follow-up PR, `git mv` the `mne/io/_*.py` to `mne/io/*.py` (doing it in one PR makes a horrible diff!)

Helpful for #11838 but cleans things up in its own right.

### CODE REVIEW

File: doc/file_io.rst [hunk 1]
```diff
@@ -54,6 +54,7 @@ File I/O
    what
    io.read_info
    io.show_fiff
+   io.pick.get_channel_type_constants
```
Comment (larsoner) [inline doc/file_io.rst:57]: This was documented as a public attribute in `whats_new.rst` so I guess we live with it :shrug:

Comment (drammock) [reply doc/file_io.rst:57]: we can always deprecate. hard to believe it would be very useful to end-users.

**Comment by drammock** [comment]
FYI, I added autoreject to the list of sibling packages to check.

File: mne/io/diff.py [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-#
-# License: BSD Style.
-
-import numpy as np
-
-from ..utils import logger, verbose
-
-
-@verbose
-def is_equal(first, second, verbose=None):
```
Comment (larsoner) [inline mne/io/diff.py:11]: This was intentionally removed. I had never seen it before, it was never used anywhere AFAIK

File: doc/file_io.rst (hunk 1 above)
Comment (larsoner) [reply doc/file_io.rst]: Okay, done -- moved to `mne.io.get_channel_type_constants` (it's implicitly deprecated via the magic `__getattr__`)

File: mne/io/_pick.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    _DATA_CH_TYPES_SPLIT,
+)
+
+__all__ = [
+    # mne-bids, autoreject, mne-connectivity, mne-realtime, mne-nirs, mne-realtime
+    "_picks_to_idx",
+    # mne-qt-browser
+    "_DATA_CH_TYPES_ORDER_DEFAULT",
+    "_DATA_CH_TYPES_SPLIT",
+]
```
Comment (larsoner) [inline mne/io/_pick.py:19]: @drammock other than `mne.io.constants.FIFF` (which is pretty painless to keep) this is the list of what we have to keep around to make submodules happy. Do you think we should make some variant of `_picks_to_idx` public?

Comment (drammock) [reply mne/io/_pick.py:19]: yes I do.  I notice it was used in MNE-BIDS and it's a pretty common thing to want to do (I believe @Aaronearlerichardson mentioned something about it in #11899 today in fact)

**Comment by larsoner** [comment]
Okay to merge @drammock ?

**Pull request merged**